pilot/pkg/model/push_context.go

```diff
@@ -1746,15 +1760,19 @@ func (ps *PushContext) initVirtualServices(env *Environment) {
 				if gw == constants.IstioMeshGateway {
 					continue
 				}
-				for host := range virtualServiceDestinations(rule) {
+				gatewayNamespace := ns
+				if gwNs, _, found := strings.Cut(gw, "/"); found {
+					gatewayNamespace = gwNs
```

Comment:
I am sure this is not right, a `gateway` CR can reside in different namesapce as the gateway isntance.

Reply:
yeah i see it would not work. and IMO for sidecar it also applies to the namespace of sidecar instead of namespace of instance

Reply:
There is also an option to apply this filtering logic only when the PILOT_SCOPE_GATEWAY_TO_NAMESPACE is enabled.

Reply:
Agree, the legacy behavior is a little tricky

Reply:
I pushed a change which filters these destinations when the  PILOT_SCOPE_GATEWAY_TO_NAMESPACE is enabled, otherwise it behaves as before

Reply:
@hzxuzhonghu can you rereview it?